lib/src/command/dependency_services.dart

```diff
@@ -457,8 +463,44 @@ class DependencyServicesApplyCommand extends PubCommand {
           writeTextFile(entrypoint.pubspecPath, updatedPubspec);
         }
         // Only if we originally had a lock-file we write the resulting lockfile back.
-        if (lockFileEditor != null) {
-          entrypoint.saveLockFile(solveResult);
+        if (updatedLockfile != null) {
+          final updatedPackages = <PackageId>[];
+          for (final package in solveResult.packages) {
+            if (package.isRoot) continue;
+            final description = package.description;
+            if (description is ResolvedHostedDescription &&
+                description.sha256 == null) {
+              // We removed the hash above before resolution - as we get the
+              // locked id back we need to find the content-hash from the
+              // version listing.
+              //
+              // `pub get` gets this version-listing from the downloaded archive
+              // but we don't want to download all archives - so we copy it from
+              // the version listing.
+              var listedId = (await cache.getVersions(package.toRef()))
+                  .firstWhere((id) => id == package);
```

Comment:
Not sure if this is wise, it should never be necessary.
```suggestion
              var listedId = (await cache.getVersions(package.toRef()))
                  .firstWhere((id) => id == package, orElse: () => package);
```
I'm just thinking the error we get from trying to download it, might be better :D